packages/Swiper/index.tsx

```diff
@@ -1,12 +1,55 @@
-import React, { cloneElement, forwardRef, useEffect, useState } from 'react'
-import { bool, func, node, number, string } from 'prop-types'
+import React, { cloneElement, forwardRef, ReactNode, useEffect, useState } from 'react'
 import { useTheme } from '@xstyled/styled-components'
 import { useViewportSize } from '@welcome-ui/utils'
+import { WuiProps } from '@welcome-ui/system'
 
 import { useInterval } from './use-interval'
 import * as S from './styles'
 
-export const Swiper = forwardRef((props, ref) => {
+export interface RenderPaginationProps {
+  idx: number
+  'aria-controls': React.HTMLAttributes<HTMLElement>['aria-controls']
+  'aria-label': React.HTMLAttributes<HTMLElement>['aria-label']
+  'aria-selected': React.HTMLAttributes<HTMLElement>['aria-selected']
+  onClick: () => void
+  pageIdx: number
+}
+
+export interface UseSwiperProps {
+  autoplay?: boolean
+  duration?: number
+  loop?: boolean
+  slidesToShow?: number
+  slidesToSwipe?: number
+  nextButton?: ReactNode
+  prevButton?: ReactNode
+}
+
+export interface UseSwiperState {
+  goNext: React.MouseEventHandler<HTMLDivElement>
+  goPrev: React.MouseEventHandler<HTMLDivElement>
+  loop: boolean
+  numberOfSlides: number
+  pageIdx: number
+  setNumberOfSlides: (slidesLength: number) => void
+  setPageIdx: (idx: number) => void
+  slidesToShow: number
+  slidesToSwipe: number
+  nextButton?: ReactNode
+  prevButton?: ReactNode
+}
+
+export interface SwiperOptions {
+  renderPaginationItem?: (props: RenderPaginationProps) => ReactNode
```

Comment:
We need to be same everywhere on Types import, React.ReactNode or ReactNode ?

Reply:
React.ReactNode is better I guess

Reply:
done !